ui/src/terraui/build.sh

```diff
@@ -13,8 +13,8 @@ cd "$WD"
 REPODIR=.repo
 
 if [[ -e $REPODIR ]]; then
-  set +e; (cd $REPODIR; git log $TERRA_UI_GIT_HASH..$TERRA_UI_GIT_HASH); set -e
-  if [ $? -eq 0 ]; then
+  gitlogexitcode=$(cd $REPODIR; git log $TERRA_UI_GIT_HASH..$TERRA_UI_GIT_HASH>/dev/null; echo $?)
+  if [[ $gitlogexitcode == 0 ]]; then
```

Comment:
This is covered in [the guide](http://redsymbol.net/articles/unofficial-bash-strict-mode/), but the synopsis is that `set -e` will cause the script to exit when any command fails. The above `git log` will fail if the specified hash isn't found, but we don't want to exit the script. This is an expected failure, so we want to provide an error message.

This line:
```
gitlogexitcode=$(cd $REPODIR; git log $TERRA_UI_GIT_HASH..$TERRA_UI_GIT_HASH>/dev/null; echo $?)
```
will succeed because `echo $?` will succeed. Redirecting stdout to `/dev/null` ensures that only the exit code will appear in the output.